custom/icds_reports/views.py

```diff
@@ -2164,3 +2170,43 @@ def get(self, request, *args, **kwargs):
         except Exception:
             response = dict(isSuccess=False, message='Unknown Error occured')
             return JsonResponse(response, status=500)
+
+
+@location_safe
+class BaseGovernanceAPIView(View):
```

Comment:
Why do you need this base class?

Reply:
Here we are decoding all the request params and settings dates. Wrote it by a reference from the earlier apis which has a baseclass to prepare request params

Reply:
I think others have a base class as same base class could be used by other sub classes. Here it looks its only extended by only one class. I think this maybe be not needed